tools/cellprofiler/common.xml

```diff
@@ -0,0 +1,441 @@
+
+    _str += FOURSPACES + "Do you want to group your images?:%s\n" % groups["con_groups"]["group_images"]
+    _str += FOURSPACES + "grouping metadata count:1\n"
+    _str += FOURSPACES + "Metadata category:%s\n" % groups["con_groups"]["group_category"]
+
+    return _str
+
+
+with open("output", "w") as f:
+    headers = ["CellProfiler Pipeline: http://www.cellprofiler.org\n",
+               "Version:3\n",
+               "DateRevision:319\n",
+               "GitHash:\n",
+               "ModuleCount:4\n",
+               "HasImagePlaneDetails:False",
+               "\n"]
+
+    f.writelines(headers)
+
+    img_str = write_images()
+    metadata_str = write_metadata()
+    nameandtypes_str = write_nameandtypes()
+    groups_str = write_groups()
+
+    output_str = "%s%s%s%s" %(img_str, metadata_str, nameandtypes_str, groups_str)
+
+    f.write(output_str)
+    f.close()
+
+        </configfile>
+    </configfiles>
+    <inputs>    
+<!-- Images module-->
+        <section name="images" title="Images" expanded="false">
+            <param name="filter_images" type="select" label="Filer images?">
+              <option value="Images only">Images only</option>
+              <option value="No filtering">No filtering</option>
+            </param>
+        </section>
+    <!-- Metadata module-->
+        <section name="metadata" title="Metadata" expanded="false">
+            <conditional name="con_metadata_extraction">
+                <param name="extract" type="select" display="radio" label="Extract metadata?">
+                    <option value="No">No</option>
+                    <option value="Yes">Yes</option>
+                </param>
+                <when value="Yes">
+                    <repeat name="extraction_method" title="metadata">
+                        <param name="metadata_extraction_method" type="select" label="Metadata extraction method">
+                            <option value="Extract from file/folder names">Extract from file/folder names</option>
+                            <option value="Import from file">Import from file</option>
+                            <option value="Extract from image file headers">Extract from image file headers</option>
+                        </param>
+                        <param name="metadata_source" type="select" label="Metadata source">
+                            <option value="File name">File name</option>
+                            <option value="Folder name">Folder name</option>
+                        </param>
+                        <param name="file_name_regex" type="text" value="^(?P&lt;Plate>.*)_(?P&lt;Well>[A-P][0-9]{2})_s(?P&lt;Site>[0-9])_w(?P&lt;ChannelNumber>[0-9])" label="Regular expression to extract from file name">    

```

Comment:
Not sure if this is left:
```
<param name="file_name_regex" type="select" label="Regular expression to extract from file name">
	<option value="(?P<A>.*)">A</option>
	<option value="(?P<A>.*)-(?P<B>[0-9]+)">A-B</option>
	<option value="(?P<A>.*)_(?P<B>[0-9]+)">A_B</option>
	<option value="(?P<A>.*)__(?P<B>[0-9]+)">A__B</option>
	<option value="(?P<A>.*)-(?P<B>[0-9]+)-(?P<C>[0-9]+)">A-B-C</option>
	<option value="(?P<A>.*)_(?P<B>[0-9]+)_(?P<C>[0-9]+)">A_B_C</option>
	<option value="(?P<A>.*)__(?P<B>[0-9]+)__(?P<C>[0-9]+)">A__B__C</option>
	<option value="(?P<A>.*)-(?P<B>[0-9]+)-(?P<C>[0-9]+)-(?P<D>[0-9]+)">A-B-C-D</option>
	<option value="(?P<A>.*)_(?P<B>[0-9]+)_(?P<C>[0-9]+)_(?P<D>[0-9]+)">A_B_C_D</option>
	<option value="(?P<A>.*)__(?P<B>[0-9]+)__(?P<C>[0-9]+)__(?P<D>[0-9]+)">A__B__C__D</option>
	<option value="(?P<A>.*)-(?P<B>[0-9]+)-(?P<C>[0-9]+)-(?P<D>[0-9]+)-(?P<E>[0-9]+)">A-B-C-D-E</option>
	<option value="(?P<A>.*)_(?P<B>[0-9]+)_(?P<C>[0-9]+)_(?P<D>[0-9]+)_(?P<E>[0-9]+)">A_B_C_D_E</option>
	<option value="(?P<A>.*)__(?P<B>[0-9]+)__(?P<C>[0-9]+)__(?P<D>[0-9]+)__(?P<E>[0-9]+)">A__B__C__D__E</option>
	<option value="(?P<A>.*)-(?P<B>[0-9]+)-(?P<C>[0-9]+)-(?P<D>[0-9]+)-(?P<E>[0-9]+)-(?P<F>[0-9]+)">A-B-C-D-E-F</option>
	<option value="(?P<A>.*)_(?P<B>[0-9]+)_(?P<C>[0-9]+)_(?P<D>[0-9]+)_(?P<E>[0-9]+)_(?P<F>[0-9]+)">A_B_C_D_E_F</option>
	<option value="(?P<A>.*)__(?P<B>[0-9]+)__(?P<C>[0-9]+)__(?P<D>[0-9]+)__(?P<E>[0-9]+)__(?P<F>[0-9]+)">A__B__C__D__E__F</option>
```

Reply:
just to clarify again..
- we dont expect users to replace the A, B, C, D E, F with something else, the regex is used as is, ?
- can I please have 1 or 2 file name examples that is known to work based on any one of rhte patterns above please?

Reply:
Yeah, I think it would be nice to have real-example file names as part of the help section.

Reply:
I will provide those examples :)

Reply:
is this for the tests?